client/packages/requisitions/src/RequestRequisition/ListView/AppBarButtons.tsx

```diff
@@ -49,13 +53,55 @@ export const AppBarButtons: FC<{
     exportCSV(csv, t('filename.requests'));
   };
 
+  // Warning when you don't have a recent stocktake with enough items
+  const prefs = usePreferences();
+
+  const {
+    query: { data: recentStocktakeData, isLoading: stocktakeLoading },
+  } = useRecentStocktakes(
+    prefs.warnWhenMissingRecentStocktake?.enabled ?? false,
+    prefs.warnWhenMissingRecentStocktake?.maxAge
+  );
+
+  const uniqueStocktakeItemCount =
+    getUniqueItemCountInStocktakeLines(recentStocktakeData);
+
+  // Determine whether to show the stocktake too old warning
+  const showOldStocktakeWarning =
+    prefs.warnWhenMissingRecentStocktake?.enabled &&
+    prefs.warnWhenMissingRecentStocktake?.minItems !== undefined &&
+    uniqueStocktakeItemCount < prefs.warnWhenMissingRecentStocktake?.minItems;
+
+  const getConfirmation = useConfirmationModal({
+    message: t('warning.insufficient-recent-stocktake-items', {
+      minItems: prefs.warnWhenMissingRecentStocktake?.minItems,
+      maxAge: prefs.warnWhenMissingRecentStocktake?.maxAge,
+    }),
+    title: t('heading.are-you-sure'),
```

Comment:
I'm not sure I totally like how it auto-navigates you to Stocktakes when clicking "Cancel", although I see the point. I think at the very least we should change the Cancel button text to something like "Go to Stocktakes" so you at least know what's happening.

Can do this with the `cancelButtonLabel` prop